python/mxnet/rnn/rnn_cell.py

```diff
@@ -117,21 +118,29 @@ def reset(self):
         self._counter = -1
 
     def __call__(self, inputs, states):
-        """Construct symbol for one step of RNN.
+        """Unroll the RNN for one time step.
 
         Parameters
         ----------
         inputs : sym.Variable
             input symbol, 2D, batch * num_units
-        states : sym.Variable
-            state from previous step or begin_state().
+        states : list of sym.Variable
+            RNN state from previous step or the output of begin_state().
 
         Returns
         -------
         output : Symbol
-            output symbol
-        states : Symbol
-            state to next step of RNN.
+            output symbol on unrolling this time step.
+        states : list of Symbols
+            states symbol is the new state of this RNN after this unrolling. 
+            The type of this is same as the output of begin_state().
+            This can be used as input state to the next timestep 
+            of this RNN.
+
+        See Also
+        --------
+        begin_state: This function can provide the states for the first time step.
```

Comment:
You can also override the first time step states yourself and not use begin_state(). This is why I used "can". I don't feel too strongly one way or the other on this one though.